src/applications/personalization/profile-2/components/direct-deposit/DirectDepositContent.jsx

```diff
@@ -0,0 +1,311 @@
+  directDepositInformation,
+  directDepositIsSetUp,
+  directDepositUiState as directDepositUiStateSelector,
+} from 'applications/personalization/profile360/selectors';
+
+import BankInfoForm from 'applications/personalization/profile360/components/BankInfoForm';
+import PaymentInformationEditError from 'applications/personalization/profile360/components/PaymentInformationEditModalError';
+
+import ProfileInfoTable from '../ProfileInfoTable';
+import FraudVictimAlert from './FraudVictimAlert';
+import AdditionalInformation from './DirectDepositInformation';
+
+import { prefixUtilityClasses } from '../../helpers';
+
+// TODO: move this to a common location (in platform?) for easy sharing
+/**
+ * While not included with React, this hook is described in the React Hooks
+ * documentation:
+ * https://reactjs.org/docs/hooks-faq.html#how-to-get-the-previous-props-or-state
+ *
+ * as well as on usehooks.com:
+ * https://usehooks.com/usePrevious/
+ *
+ * This post does a deeper dive for those who are new to hooks:
+ * https://blog.logrocket.com/how-to-get-previous-props-state-with-react-hooks/
+ */
+const usePrevious = value => {
+  const valueRef = useRef();
+  useEffect(
+    () => {
+      valueRef.current = value;
+    },
+    // only run this useEffect hook if `value` changes
+    [value],
+  );
+  // this returns _before_ the `useEffect` call completes, so this will return
+  // whatever was stored in `ref.current` _before_ this function's `useEffect`
+  // callback runs
+  return valueRef.current;
+};
+
+export const DirectDepositContent = ({
+  isAuthorized,
+  isDirectDepositSetUp,
+  directDepositAccountInfo,
+  directDepositUiState,
+  saveBankInformation,
+  toggleEditState,
+}) => {
+  const editBankInfoButton = useRef();
+  const [formData, setFormData] = useState({});
+  const [showSaveSucceededAlert, setShowSaveSucceededAlert] = useState(false);
+  const previousUiState = usePrevious(directDepositUiState);
+
+  const isEditingBankInfo = directDepositUiState.isEditing;
+
+  // when we enter and exit edit mode...
+  useEffect(
+    () => {
+      // reset the bank info form when we enter edit mode
+      if (isEditingBankInfo) {
+        setFormData({});
+      }
+      // focus the edit button when we exit edit mode
+      if (isEditingBankInfo === false) {
+        editBankInfoButton.current.focus();
+      }
+    },
+    [isEditingBankInfo],
+  );
+
+  // show the user an alert after their bank info has been updated and
+  useEffect(
+    () => {
+      if (
+        // TODO: can we add something in Redux that will flip to true after a
+        // save has succeeded and then goes back to false when we are editing
+        // bank info again?
+        previousUiState?.isSaving &&
+        !directDepositUiState.isSaving &&
+        !directDepositUiState.responseError
+      ) {
+        setShowSaveSucceededAlert(true);
+        setTimeout(() => {
+          setShowSaveSucceededAlert(false);
+        }, 6000);
```

Comment:
This could be a constant somewhere too.

I wonder if it'd be worthwhile to either add this to the timeouts in src/platform/testing (maybe not, since those are for testing....) or, create a set of timeout constants somewhere in platform that can be used throughout different applications.

Reply:
Yeah, perhaps. But I really did this one by feel. How long you show an alert before hiding it really depends on the amount of text in the alert so that isn't a great fit for a shared constant.